backend/src/ml_space_lambda/authorizer/lambda_function.py

```diff
@@ -176,31 +174,34 @@ def lambda_handler(event, context):
                 # Path params need to be checked individually
                 elif "projectName" in path_params:
                     project_name = path_params["projectName"]
-                    project_user = project_user_dao.get(project_name, username)
                     # User must belong to the project for any project specific resources
-                    if project_user or Permission.ADMIN in user.permissions:
+                    if IS_ADMIN or is_member_of_project(user.username, project_name):
+                        IS_OWNER = is_owner_of_project(user.username, project_name)
+                        project_user = project_user_dao.get(project_name, username)
                         # User must be an owner or admin to add/remove users or update the project config
                         if (
                             (
                                 request_method == "POST"
-                                and (requested_resource.endswith("/users") or requested_resource.endswith("/app-config"))
+                                and (
+                                    requested_resource.endswith("/users")
+                                    or requested_resource.endswith("/groups")
+                                    or requested_resource.endswith("/app-config")
+                                )
                             )
-                            or (request_method in ["PUT", "DELETE"] and len(path_params) == 2 and "username" in path_params)
-                        ) and (
-                            (project_user and Permission.PROJECT_OWNER not in project_user.permissions)
-                            and Permission.ADMIN not in user.permissions
-                        ):
+                            or (
+                                request_method in ["PUT", "DELETE"]
+                                and len(path_params) == 2
+                                and ("username" in path_params or "groupName" in path_params)
```

Comment:
Yeah, this is disgusting. 

Would really like to see about adding this in the individual `lambda_functions.py` files or maybe we break this out so that we are handling each entity in it's own method. Having groups combined with users is just going to lead some bugs.

Can be a future enhancement but let's get something in the backlog.